### DISCUSSION TITLE
blog/nasa-power-and-zarr/

### DISCUSSION THREAD

**Discussion by giscus** [zarr-developers/blog#58] [General]
# blog/nasa-power-and-zarr/

Blog post on Zarr usage at NASA LARC’s POWER project

https://zarr.dev/blog/nasa-power-and-zarr/

**Comment by gaiadhi** [comment]
Hi
We are transitioning from NASA Power REST to Zarr, mainly as a speedup.  But have run into inconsistencies with precipitation and solar radiation. 
1. Precipitation
a. We use s3://power-analysis-ready-datastore/power_901_daily_meteorology_utc.zarr, and extract PRECTOTCORR, which we believe is in kg/m2/s
b. When hitting https://power.larc.nasa.gov/api/temporal/daily/point?parameters=PRECTOTCORR&community=ag&longitude={longitude}&latitude={latitude}&start=20100101&end=20191231&format=JSON, we get in mm/day
c. Multiplying the Zarr data by 86400 should give us the same value as in b); but we see quite a bit of deviation.  STD is 4. 
2. Solar radiation 
a. We use s3://power-analysis-ready-datastore/power_901_daily_radiation_utc.zarr and extract ALLSKY_SFC_LW_DWN, which we believe is in W m-2.
b. When hitting https://power.larc.nasa.gov/api/temporal/daily/point?parameters=ALLSKY_SFC_SW_DWN&community=ag&longitude={longitude}&latitude={latitude}&start=20100101&end=20191231&format=JSON, we get srad in MJ/m^2/day
c. Converting the Zarr data (/= (1000000 / 86400)) does not at all match up with what we get in b).

Can you guide us to how to minimize the discrepancies??
Thanks
Karthik

**Reply by NASA-LaRC-POWER-Project** [reply]
Great to hear you are migrating from the POWER API to the Zarr for direct access to the POWER Datastores. The difference you may be seeing is that the API's do additional data value checking to remove any erroneous values before sending a response. The API's use the POWER Zarr datastores so the values should be identical aside from the unit conversions that the API's provide based on community and previously mentions additional checks. Lastly, when accessing the Zarr you are not limited to the level of precision provided by the API so please be sure to check the valid ranges for the specific data parameters.

One discrepancy we can identify in the example provided is the corresponding Zarr datastore selected is in Universal Time Coordinated (UTC) time not the API’s default time standard of Local Solar Time (LST).

To confirm we have listed the conversions to your requested units:
- "W m-2" * 0.0864 = "MJ/m^2/day"
- "kg m-2 s-1" * 86400 == "mm"

If you have more questions, please email us at [larc-power-project@mail.nasa.gov](larc-power-project@mail.nasa.gov) or review the [POWER Docs](https://power.larc.nasa.gov/docs/).
